samples/matter/common/src/Kconfig

```diff
@@ -40,18 +40,47 @@ config NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE
 	  Allow device to perform factory reset if the operational key for Fabric has not been migrated
 	  properly to PSA ITS storage.
 
-config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+menuconfig NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
 	bool "Enable diagnostics logs support"
 	help
 	  Provides support for diagnostics logs. Diagnostics logs allow the Matter controller to read end-user,
 	  network and crash logs from the Matter device.
 	  Diagnostic logs are usable only if the DiagnosticCluster is available and active on the device's root endpoint.
 	  Otherwise the compilation error occurs.
 
+if NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+
 config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_MAX_SIMULTANEOUS_SESSIONS
 	int "Define maximum simultaneous sessions"
 	default 20
 	help
 	  The maximum simultaneous sessions, which is the capability for sending diagnostic logs
 	  to Matter controller using multiple sessions at the same time for different diagnostic
 	  intents. When a session finishes, it releases its slot for a new one.
+
+config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS
+	bool "Enable crash logs support"
```

Comment:
nit: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#id22 discourages adding redundant "Enable" in Kconfig prompts. It's clear from the type that it enables something.